### ISSUE TITLE
Support for setting maintenance windows on azurerm_mssql_database

### ISSUE THREAD

**Issue by jkelley23** [hashicorp/terraform-provider-azurerm#14520]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In the Azure Portal, you are able to set the Maintenance Window for SQL Databases in the "Maintenance" blade under "Settings." We would like to change the window from the system default for all of our Azure SQL databases. It would be great if we were able to set this in Terraform in lieu of switching it manually for every SQL database in the portal.

### New or Affected Resource(s)

* azurerm_mssql_database

### Potential Terraform Configuration

```hcl
resource "azurerm_mssql_database" "example" {
  name           = "example-sql-db"
  server_id      = azurerm_mssql_server.exampleserver.id
  collation      = "SQL_Latin1_General_CP1_CI_AS"
  license_type   = "LicenseIncluded"
  max_size_gb    = 4
  read_scale     = true
  sku_name       = "BC_Gen5_2"
  zone_redundant = true
  maintenance_configuration = ("SQL_CentralUS_DB_1", "SQL_CentralUS_DB_2", or "SQL_Default")

  extended_auditing_policy {
    storage_endpoint                                        = azurerm_storage_account.example.primary_blob_endpoint
    storage_account_access_key                       = azurerm_storage_account.example.primary_access_key
    storage_account_access_key_is_secondary = true
    retention_in_days                                        = 6
  }
```

**Comment by SachinKumar2001** [comment]
This feature will help to minimize impact on the application relying on SQL DB during business hours.

**Comment by neilmca-inc** [comment]
Can here once I found I couldn't do this. Is it on a roadmap?

**Comment by neilmca-inc** [comment]
Supported in the Azure CLI for both new provisions and existing

https://docs.microsoft.com/en-gb/azure/azure-sql/database/maintenance-window-configure?view=azuresql&tabs=azure-cli

**Comment by jkelley23** [comment]
Still waiting on an update from the @HashiCorp team.

**Comment by peterprumbach-telekom** [comment]
As mentioned above the pull-request adding this feature has been merged and it is available in version v3.22.0. You can use the `maintenance_configuration_name` to set your configuration. I think the issue can be closed.

**Comment by waynejnicklin** [comment]
In the help for SQL database, it says:

maintenance_configuration_name - (Optional) The name of the Public Maintenance Configuration window to apply to the database.

So that only allows you to choose the publicly available ones listed in the help, not a custom one that we define ourselves using the terraform maintenance_configuration resource?

**Comment by bartduncan** [comment]
> So that only allows you to choose the publicly available ones listed in the help, not a custom one that we define ourselves using the terraform azurerm_maintenance_configuration resource?

@waynejnicklin this is an Azure SQL DB limitation, not something imposed by the Terraform implementation. The Azure SQL DB service does not currently support performing maintenance at a different custom time for each resource. A fixed set of "public" maintenance configurations are available -- these have a "DB_1" (weekday evenings) or "DB_2" (weekend evenings) suffix. More information at [Maintenance window in Azure SQL Database](https://learn.microsoft.com/en-us/azure/azure-sql/database/maintenance-window?view=azuresql)

**Comment by waynejnicklin** [comment]
We set the maintenance schedule currently using REST API, and that works fine, so it's definitely possible. I just wanted to remove that post-deployment step and do it via the Terraform configuration instead.

**Comment by katbyte** [comment]
Thank you for the report, and apologies for the slow follow up here.

This issue is linked to #18247 (`azurerm_mssql_database` - add support for maintenance_configuration_name), which merged and shipped in v3.22.0 and appears to have fixed it, so we are closing this one out as resolved.

If you are still running into this on the current (v5.x) version of the provider, please let us know with a comment or open a new issue referencing this one and we will take another look.
